strawberryfields/engine.py

```diff
@@ -661,33 +661,30 @@ def run_async(
                     f"to compile with {compiler_name}."
                 )
 
-        if program.type != "tdm":
```

Comment:
Hmm, I'm getting the following error for a remote job, if this `if` statement is removed :thinking::
```python
/strawberryfields/strawberryfields/engine.py in run(self, program, compile_options, recompile, **kwargs)
    589         """
    590         job = self.run_async(
--> 591             program, compile_options=compile_options, recompile=recompile, **kwargs
    592         )
    593         try:

/strawberryfields/strawberryfields/engine.py in run_async(self, program, compile_options, recompile, **kwargs)
    689         # update the run options if provided
    690         run_options = {}
--> 691         run_options.update(program.run_options)
    692         run_options.update(kwargs or {})
    693 

AttributeError: 'NoneType' object has no attribute 'run_options'
```

Reply:
This was two-fold: `TDMProgram.compile` wasn't returning the compiled program (which it doesn't really need to since it changes nothing), but for compatibility I set it to return `self`; and also there was no `TDMprogram.run_options` dict, which I also added since it actually is needed both here and for the Blackbird to TDMProgram conversion.